### PR TITLE
Split default/custom Container Apps.

### PR DESCRIPTION
**Pull request by PsypherPunk**
### Context

The current `data-pipeline-job-start` queue will instead be split into `data-pipeline-job-custom-start` and `data-pipeline-job-default-start` queues, each serviced by a distinct Container App.

[AB#235428](https://dfe-ssp.visualstudio.com/a14e55df-4fbf-4a2f-a11d-22b187178343/_workitems/edit/235428)

### Change proposed in this pull request

- add the new queues
- refactor the existing `container_apps.tf` into a Terraform module
  - call this module twice, once for `default` and once for `custom` apps.

### Guidance to review 

WIP

### Checklist (add/remove as appropriate)

- [ ] Work items have been linked [(use AB#)](https://learn.microsoft.com/en-us/azure/devops/boards/github/link-to-from-github?view=azure-devops#use-ab-to-link-from-github-to-azure-boards-work-items)
- [ ] Your code builds clean without any errors or warnings
- [ ] You have run all unit/integration tests and they pass
- [ ] Your branch has been rebased onto main
- [ ] You have tested by running locally
- [ ] You have reviewed with UX/Design

### CODE REVIEW

File: data-pipeline/terraform/container_app/variables.tf [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+  description = "Prefix to be used for resources in the current environment."
+  type        = string
+}
+
+variable "worker-queue-name" {
+  description = "Name of the queue which will trigger the worker."
+  type        = string
+}
+
+variable "db-host" {
```
Comment (PsypherPunk) [inline data-pipeline/terraform/container_app/variables.tf:26]: TODO: something more descriptive, natch.

File: data-pipeline/terraform/container_apps.tf [hunk 2]
```diff
@@ -14,136 +14,50 @@ resource "azurerm_container_app_environment" "main" {
...
+  environment        = var.environment
+  environment-prefix = var.environment-prefix
 
-  registry {
-    server               = data.azurerm_container_registry.acr.login_server
-    username             = data.azurerm_container_registry.acr.admin_username
-    password_secret_name = "registry-password"
-    #     identity = azurerm_user_assigned_identity.container-app.id
-  }
+  db-host      = data.azurerm_key_vault_secret.core-db-domain-name.value
```
Comment (PsypherPunk) [inline data-pipeline/terraform/container_apps.tf:30]: Should values like this be pulled in via `data` references in the module? Or passed in here?

This has a clearer API for the module but I need to check whether this exposes more sensitive info. in Terraform state.

File: data-pipeline/terraform/container_apps.tf [hunk 3]
```diff
@@ -14,136 +14,50 @@ resource "azurerm_container_app_environment" "main" {
...
-    #     identity_ids = [azurerm_user_assigned_identity.container-app.id]
-  }
+module "container_app_default" {
+  source = "./container_app"
 
-  secret {
-    name  = "queue-connection-string"
-    value = data.azurerm_storage_account.main.primary_connection_string
-  }
+  container-app-environment-id = azurerm_container_app_environment.main.id
```
Comment (PsypherPunk) [inline data-pipeline/terraform/container_apps.tf:20]: Need to clarify expectations between `azurerm_container_app_environment` and `azurerm_container_app` in the context of the new Apps. running: should each Container App share the same Container App Environment? Or each have a unique one?

File: data-pipeline/terraform/container_apps.tf (hunk 2 above)
Comment (PsypherPunk) [reply data-pipeline/terraform/container_apps.tf]: Moved most of the sensitive values to `data` sources, passing in the key-vault details and secret-names.

File: data-pipeline/terraform/data.tf [hunk 4]
```diff
@@ -13,32 +8,7 @@ data "azurerm_log_analytics_workspace" "application-insights-workspace" {
   resource_group_name = "${var.environment-prefix}-ebis-core"
 }
 
-data "azurerm_container_registry" "acr" {
-  name                = "${var.environment-prefix}acr"
-  resource_group_name = "${var.environment-prefix}-ebis-core"
-}
-
 data "azurerm_storage_account" "main" {
```
Comment (PsypherPunk) [inline data-pipeline/terraform/data.tf:11]: Hmmm…as the resulting `primary_connection_string` is sensitive, this should probably come out too.

File: data-pipeline/terraform/container_apps.tf (hunk 3 above)
Comment (PsypherPunk) [reply data-pipeline/terraform/container_apps.tf:20]: Discussed with @jrabbott and will leave this as-is for now.

**Review by jrabbott — APPROVED** [review]

**Pull request merged**